src/store/auth.js

```diff
@@ -1,178 +1,245 @@
-          return this.roarfirekit.initiateLoginWithEmailLink({ email, redirectUrl }).then(() => {
-            window.localStorage.setItem('emailForSignIn', email);
-          });
-        }
-      },
-      async signInWithEmailLink({ email, emailLink }) {
-        if (this.isFirekitInit) {
-          return this.roarfirekit.signInWithEmailLink({ email, emailLink }).then(() => {
-            window.localStorage.removeItem('emailForSignIn');
-          });
-        }
-      },
-      async signInWithGooglePopup() {
-        if (this.isFirekitInit) {
-          return this.roarfirekit.signInWithPopup(AUTH_SSO_PROVIDERS.GOOGLE);
-        }
-      },
-      async signInWithGoogleRedirect() {
-        return this.roarfirekit.initiateRedirect(AUTH_SSO_PROVIDERS.GOOGLE);
-      },
-      async initStateFromRedirect() {
-        this.spinner = true;
-        const enableCookiesCallback = () => {
-          const router = useRouter();
-          router.replace({ name: 'EnableCookies' });
-        };
-        if (this.isFirekitInit) {
-          return await this.roarfirekit.signInFromRedirectResult(enableCookiesCallback).then((result) => {
-            // If the result is null, then no redirect operation was called.
-            if (result !== null) {
-              this.spinner = true;
-            } else {
-              this.spinner = false;
-            }
-          });
-        }
-      },
-      async forceIdTokenRefresh() {
-        await this.roarfirekit.forceIdTokenRefresh();
-      },
-      async sendMyPasswordResetEmail() {
-        if (this.email) {
-          return await this.roarfirekit.sendPasswordResetEmail(this.email).then(() => {
-            return true;
-          });
+export const useAuthStore = defineStore(
+  'authStore',
+  () => {
+    const adminAuthStateListener = ref(null);
+    const adminOrgs = ref(null);
+    const firebaseUser = ref({ adminFirebaseUser: null });
+    const roarfirekit = ref(null);
+    const routeToProfile = ref(false);
+    const showOptionalAssessments = ref(false);
+    const showSideBar = ref(false);
+    const spinner = ref(false);
+    const ssoProvider = ref(null);
+    const userClaims = ref(null);
+    const userData = ref(null);
+
+    function $reset() {
+      adminAuthStateListener.value = null;
+      adminOrgs.value = null;
+      firebaseUser.value = { adminFirebaseUser: null };
+      roarfirekit.value = null;
+      routeToProfile.value = false;
+      showOptionalAssessments.value = false;
+      showSideBar.value = false;
+      spinner.value = false;
+      ssoProvider.value = null;
+      userClaims.value = null;
+      userData.value = null;
+    }
+
+    function uid() {
+      return firebaseUser.value.adminFirebaseUser?.uid;
+    }
+
+    function roarUid() {
+      return userClaims.value?.claims?.roarUid;
+    }
+
+    function email() {
+      return firebaseUser.value.adminFirebaseUser?.email;
+    }
```

Comment:
Like above, it would be helpful to have a more descriptive name like `getUserEmail`.